### PR TITLE
Test v1.1 loop dependency support

### PR DESCRIPTION
**Pull request by dneto0**
For DependencyInfinite and DependencyLength, test
that they don't require a capability to be turned on.
Also, that they are assembled, binary parsed, and disassembled
correctly.

### CODE REVIEW

File: test/BinaryToText.cpp [hunk 1]
```diff
@@ -245,7 +245,10 @@ using RoundTripInstructionsTest =
     spvtest::TextToBinaryTestBase<::testing::TestWithParam<std::string>>;
 
 TEST_P(RoundTripInstructionsTest, Sample) {
-  EXPECT_THAT(EncodeAndDecodeSuccessfully(GetParam()), Eq(GetParam()));
+  EXPECT_THAT(
+      EncodeAndDecodeSuccessfully(GetParam(), SPV_BINARY_TO_TEXT_OPTION_NONE,
+                                  SPV_ENV_UNIVERSAL_1_1),
```
Comment (dekimir) [inline test/BinaryToText.cpp]: This turns off all 1.0 testing, tho. :(

File: test/BinaryToText.cpp [hunk 2]
```diff
@@ -285,6 +288,8 @@ INSTANTIATE_TEST_CASE_P(LoopControlMasks, RoundTripInstructionsTest,
                             "OpLoopMerge %1 %2 Unroll\n",
                             "OpLoopMerge %1 %2 DontUnroll\n",
                             "OpLoopMerge %1 %2 Unroll|DontUnroll\n",
+                            "OpLoopMerge %1 %2 DependencyInfinite\n",
+                            "OpLoopMerge %1 %2 DependencyLength 8\n",
```
Comment (dekimir) [inline test/BinaryToText.cpp]: Perhaps this should go into `TextToBinary.ControlFlow.cpp`, together with other `OpLoopMerge` tests?

See 13877a3b0e9d9e3217f726178f6d26f1bf1867cc.

File: test/BinaryToText.cpp (hunk 1 above)
Comment (dneto0) [inline test/BinaryToText.cpp]: Fixed.
New commit parameterizes the BinaryToText round trip tests over both v1.0 and 1.1.
The new dependency operands to OpLoopMerge are only tested in 1.1.

File: test/BinaryToText.cpp (hunk 2 above)
Comment (dneto0) [inline test/BinaryToText.cpp]: Most of the existing tests test the assembler parts in TextToBinary.*  and the diassembler/round trip is in BinaryToText.  I'd rather not change the pattern.  Yes, there are some tests in various TextToBinary.* that are round-trip, but that's the exception, not the rule.

**Comment by dneto0** [comment]
Addressed all feedback.

**Comment by dekimir** [comment]
+2

**Comment by dneto0** [comment]
Rebased, then pushed into master.

**Pull request merged**